ckanext/cloudstorage/storage.py

```diff
@@ -199,6 +200,12 @@ def path_from_filename(self, rid, filename):
             munge.munge_filename(filename)
         )
 
+    def get_path(self, resource_id):
+        resource = get_action('resource_show')({}, {'id': resource_id})
+        filename = resource['url'].rsplit('/', 1)[-1]
```

Comment:
Malformed URLs are common, especially when migrating from other sources. We should probably ensure that the resource type is 'upload', and if the file was uploaded by ckanext-cloudstorage there should be no need to rsplit, the URL should be the original, munged filename. (https://github.com/TkTech/ckanext-cloudstorage/blob/master/ckanext/cloudstorage/storage.py#L168).

Reply:
We use AWS and multipart uploads in our instance and self.filename is None in this case as https://github.com/TkTech/ckanext-cloudstorage/blob/master/ckanext/cloudstorage/storage.py#L170 never sets anything to it. get_path api does not provide filename either as it not stored in resources.

Reply:
Actually, now that I did some testing, in our instance none of those if-else blocks are executed, and as a result, self.filename status at None